### DISCUSSION TITLE
Pass in additional arguments to object constructor

### DISCUSSION THREAD

**Discussion by DarkKnights22** [jdbi/jdbi#2426] [Q&A]
Hello,

I have a KitData class which has a constructor annotated with @JdbiConstructor:
```
    @JdbiConstructor
    public KitData(AbstractContainer container, OfflineAccount account, @ColumnName("re_rolls") int reRolls, @ColumnName("random_kits") String randomKits) {
```

How would I go about passing in these additional arguments (`container` and `account`) into my constructor when I call the method in my DAO?

DAO method:

```
    @SqlQuery("SELECT * FROM KitData WHERE player_id = :player_id")
    KitData findKitData(AbstractHGContainer container, OfflineHGAccount account, @Bind("player_id") int playerID);
```

**Comment by stevenschlansker** [comment]
Hi @DarkKnights22 , we do not currently support passing through extra arguments directly to the mapped result types.

As a workaround, I would map the query to an intermediate `KitRow` type that only has the fields from the database, and then fill out the full `KitData` objects in Java code, mixing together the row results with your fixed objects.

If you would like the ability to directly pass arguments from a query into the constructed results, that is a reasonable feature request, although I think the implementation would not be trivial.
